### PR TITLE
feat(sessions): Enforce stricter rules on query interval

### PR DESCRIPTION
**Pull request by Swatinem**
This supercedes #24093 and also adds (default off) support for minute-resolution sessions.

CC @ahmedetefy since we paired on this today.
CC @fpacifici this will (when enabled) turn on minute-granularity sessions.

### CODE REVIEW

File: src/sentry/snuba/sessions_v2.py [hunk 1]
```diff
@@ -277,6 +279,81 @@ def __init__(self, query, params):
...
+        interval_str = "one minute" if allow_minute_resolution else "one hour"
+        raise InvalidParams(
+            f"The interval has to be a multiple of the minimum interval of {interval_str}."
+        )
+
+    if interval > ONE_DAY:
+        raise InvalidParams("The interval has to be less than one day.")
+
+    if ONE_DAY % interval != 0:
+        raise InvalidParams("The interval should divide one day cleanly.")
```
Comment (markstory) [inline src/sentry/snuba/sessions_v2.py:307]:
```suggestion
        raise InvalidParams("The interval should divide one day without a remainder.")
```
🤷

**Review by markstory — APPROVED** [review]
Looks good to me.

**Comment by HazAT** [comment]
Just fyi, this broke the mobile app cc @markstory @Swatinem 😅 

We will fix it in the app
https://github.com/getsentry/sentry-mobile/issues/164

**Comment by armenzg** [comment]
Have there been discussions on testing strategies to prevent regressions?

**Comment by HazAT** [comment]
ref: https://github.com/getsentry/sentry-mobile/issues/166

**Pull request merged**